### PR TITLE
Ci 890 live blog post tracker

### PR DESCRIPTION
**Pull request by noelenwenede**
The implementation of post level tracking for live blogs was previously done on `next-article`. As part of the process of implementing post level tracking on `ft-app`, the implementation of post-tracker has been moved to the `x-live-blog-wrapper` component. With this implementation, a new property (object) has been added to `x-live-blog-wrapper` component called `postTrackerConfig` (with the following properties/fields `onEntersViewport` [callback function], `onRead` [callback function], `onError` [callbackfunction], `usePostTracker` [boolean]).

The post-tracker class creates an intersection-observer instance and observes the visibility of posts that exist in the live-blog-wrapper component. When a new post is inserted, the wrapper component dispatches an event. The post tracker class listens for this event and updates the number of posts being observed.

### CODE REVIEW

File: components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx [hunk 1]
```diff
@@ -20,45 +30,124 @@ const withLiveBlogWrapperActions = withActions({
...
+			minMillisecondsToReport: 5000,
+			returnVisibleElement: true,
+			observerUpdateEventString: 'LiveBlogWrapper.INSERT_POST',
+			liveBlogWrapperQuery: `div[data-live-blog-wrapper-id="${id}"]`,
+			liveBlogWrapper: this.props.liveBlogWrapperElementRef
+				? this.props.liveBlogWrapperElementRef.current
+				: undefined,
+			onEntersViewport: (event) => callbacks.onEntersViewport(event),
+			onRead: (event) => callbacks.onRead(event),
+			onError: (event) => callbacks.onError(event)
```
Comment (GlynnPhillips) [inline components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx:107]: **Suggestion/Feedback:** This could error if a consuming app doesn't pass these properties in as they won't be functions. I think we need some sanitising of whats passed in to make sure we can set up a post tracker

**Review by GlynnPhillips — CHANGES_REQUESTED** [review]
I have made 1 small comment - if I am miss understanding it then let me know

File: components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx (hunk 1 above)
Comment (noelenwenede) [reply components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx:107]: So I wrote that as a `sub function` to the `setup function`. The setup function does the sanitisation here https://github.com/Financial-Times/x-dash/blob/bbe48a46699f09ca60c92a3a56e89273c83f0ded/components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx#L65 and errors out here https://github.com/Financial-Times/x-dash/blob/bbe48a46699f09ca60c92a3a56e89273c83f0ded/components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx#L71.

Would you prefer the sanitisation inside the sub function? I left all sanitisation inside the setup function.

Comment (GlynnPhillips) [reply components/x-live-blog-wrapper/src/LiveBlogWrapper.jsx:107]: I knew I had probably missed something 😄 Sorry that looks fine - just hard to follow it in the PR

**Review by GlynnPhillips — APPROVED** [review]
Looks good to me - It might be worth checking with someone form the apps team that this will work in the app as I have no knowledge of how tracking works in the app

**Comment by noelenwenede** [comment]
> Looks good to me - It might be worth checking with someone form the apps team that this will work in the app as I have no knowledge of how tracking works in the app

It should work fine on the app. The call back functions only reports local visibility tracking data, the consumers would manage sending tracking data and application context.

My worry is with `next-article` and how it hydrates on the frontend. I'll do some test on it to see if it works fine.

**Review by GlynnPhillips — APPROVED** [review]
